content/en/blog/_posts/2025-01-14-devices-failure-handling/index.md

```diff
@@ -0,0 +1,403 @@
+power, making device failures commonplace and a part of the business model.
+However Kubernetes’s view on resources is still very static. The resource is
+either there or not. And if it is there, the assumption is that it will stay
+there. This assumption combined with the overall complexity of a set up, leads
+to a variety of failure modes, which we discuss here.
+
+### Understanding AI/ML workloads
+
+Generally, all AI/ML workloads need specialized hardware, have challenging
+scheduling requirements, and are expensive when idle. AI/ML workloads typically
+fall into two categories - training and inference. Here is an oversimplified
+view on those categories’ characteristics, which are different traditional,
+workloads like web services:
+
+* **Training:** These workloads are resource-intensive, often consuming entire
+  machines and running as gangs of pods. Training jobs are usually "run to
+  completion" - but that could be days, weeks or even months. Any failure in a
+  single pod can necessitate restarting the entire step across all the pods.  
+* **Inference:** These workloads are usually long-running or run indefinitely,
+  and can be small enough to consume a subset of a Node’s devices or large to
+  span multiple nodes. They often require downloading huge files with the model
+  weights.
+
+These workload types specifically break many past assumptions:
+
+{{< table caption="Workload assumptions before and now" >}}
+| Before | Now |
+| :---- | :---- |
+| Can get a better CPU and the app will work faster. | Require a **specific** device (or **class of devices**) to run. |
+| When something doesn’t work, just recreate it. | Allocation or reallocation is expensive. |
+| Any node will work. No need to coordinate between Pods. | Scheduled in a special way - devices often connected in a cross nodes topology. |
+| Each Pod can be plug-and-play replaced if failed. | Pods are a part of a larger "task". Lifecycle of an entire "task" depends on each Pod. |
+| Images are slim and easily available. | Images are big and require special handling. |
+| Long initialization can be offset by slow rollout. | Initialization may be long and should be optimized, sometimes across many Pods together. |
+| Compute nodes are commoditized and relatively inexpensive, so some idle time is acceptable. | Nodes with specialized hardware can be an order of magnitude more expensive than those without, so idle time is very wasteful. |
+{{< /table >}}
+
+The existing failure model was relying on old assumptions. It may still work for
+the new workload types, but it has limited knowledge about devices and is very
+expensive for them. In some cases, even prohibitively expensive. You will see
+more examples later in this article.
+
+### Why Kubernetes still reigns supreme
+
+This article is not going deeper into the question of "why not start fresh" for
+AI/ML workloads since they are so different from the traditional Kubernetes
```

Comment:
```suggestoin
This article is not going deeper into the question: why not start fresh for
AI/ML workloads since they are so different from the traditional Kubernetes
```